### PR TITLE
fix(curriculum): Remove additional unnecessary assert message argument from English challenges JavaScript Algorithms and Data Structures

### PR DESCRIPTION
**Pull request by RandellDawson**
- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x]  My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] All the files I changed are in the same world language (for example: only English changes, or only Chinese changes, etc.)
- [x] My changes do not use shortened URLs or affiliate links.

Related to issue #36400

This PR removes assert message arguments from `testString` for the following sub sections of the `JavaScript Algorithms and Data Structures` curriculum section.  This PR was created to remove assert message arguments which were wrapped in two single quotes instead of single quotes.  [See this discussion for more information](https://github.com/freeCodeCamp/freeCodeCamp/pull/36409#issuecomment-510219004)

* Basic Algorithm Scripting
* Debugging
* ES6
* Functional Programming
* Basic Data Structures
* Intermediate Algorithm Scripting
* Object Oriented Programming
* Regular Expressions
* JavaScript Algorithms and Data Structures Projects

### CODE REVIEW

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-algorithm-scripting/boo-who.english.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ tests:
   - text: <code>booWho([].slice)</code> should return false.
     testString: assert.strictEqual(booWho([].slice), false, '<code>booWho([].slice)</code> should return false.');
```
Comment (vkWeb) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-algorithm-scripting/boo-who.english.md]: Fix this file.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/access-property-names-with-bracket-notation.english.md [hunk 2]
```diff
@@ -27,9 +27,9 @@ We've defined a function, <code>checkInventory</code>, which receives a scanned
...
   - text: <code>checkInventory</code> is a function
-    testString: assert.strictEqual(typeof checkInventory, 'function', '<code>checkInventory</code> is a function');
+    testString: assert.strictEqual(typeof checkInventory, 'function');
   - text: 'The <code>foods</code> object should have only the following key-value pairs: <code>apples: 25</code>, <code>oranges: 32</code>, <code>plums: 28</code>, <code>bananas: 13</code>, <code>grapes: 35</code>, <code>strawberries: 27</code>'
-    testString: 'assert.deepEqual(foods, {apples: 25, oranges: 32, plums: 28, bananas: 13, grapes: 35, strawberries: 27}, ''The <code>foods</code> object should have only the following key-value pairs: <code>apples: 25</code>, <code>oranges: 32</code>, <code>plums: 28</code>, <code>bananas: 13</code>, <code>grapes: 35</code>, <code>strawberries: 27</code>'');'
+    testString: 'assert.deepEqual(foods, {apples: 25, oranges: 32, plums: 28, bananas: 13, grapes: 35, strawberries: 27});'
   - text: <code>checkInventory("apples")</code> should return <code>25</code>
     testString: assert.strictEqual(checkInventory('apples'), 25, '<code>checkInventory("apples")</code> should return <code>25</code>');
```
Comment (vkWeb) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/access-property-names-with-bracket-notation.english.md]: 3 arguments in this file.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/modify-an-array-stored-in-an-object.english.md [hunk 3]
```diff
@@ -22,7 +22,7 @@ tests:
   - text: The <code>user</code> object has <code>name</code>, <code>age</code>, and <code>data</code> keys
     testString: assert('name' in user && 'age' in user && 'data' in user, 'The <code>user</code> object has <code>name</code>, <code>age</code>, and <code>data</code> keys');
```
Comment (vkWeb) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/modify-an-array-stored-in-an-object.english.md]: 2 arguments in this file.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/modify-an-object-nested-within-an-object.english.md [hunk 4]
```diff
@@ -43,7 +43,7 @@ tests:
   - text: The <code>online</code> property nested in the <code>data</code> key of <code>userActivity</code> should be set to <code>45</code>
     testString: assert(userActivity.data.online === 45, 'The <code>online</code> property nested in the <code>data</code> key of <code>userActivity</code> should be set to <code>45</code>');
```
Comment (vkWeb) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/modify-an-object-nested-within-an-object.english.md]: 3 arguments in this file.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/use-the-delete-keyword-to-remove-object-properties.english.md [hunk 5]
```diff
@@ -27,7 +27,7 @@ Use the delete keyword to remove the <code>oranges</code>, <code>plums</code>, a
 ```yml
 tests:
   - text: 'The <code>foods</code> object only has three keys: <code>apples</code>, <code>grapes</code>, and <code>bananas</code>'
-    testString: 'assert(!foods.hasOwnProperty(''oranges'') && !foods.hasOwnProperty(''plums'') && !foods.hasOwnProperty(''strawberries'') && Object.keys(foods).length === 3, ''The <code>foods</code> object only has three keys: <code>apples</code>, <code>grapes</code>, and <code>bananas</code>'');'
+    testString: 'assert(!foods.hasOwnProperty(''oranges'') && !foods.hasOwnProperty(''plums'') && !foods.hasOwnProperty(''strawberries'') && Object.keys(foods).length === 3);'
   - text: The <code>oranges</code>, <code>plums</code>, and <code>strawberries</code> keys are removed using <code>delete</code>
     testString: assert(code.search(/oranges:/) !== -1 && code.search(/plums:/) !== -1 && code.search(/strawberries:/) !== -1, 'The <code>oranges</code>, <code>plums</code>, and <code>strawberries</code> keys are removed using <code>delete</code>');
```
Comment (vkWeb) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/use-the-delete-keyword-to-remove-object-properties.english.md]: This one here.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/delete-properties-from-a-javascript-object.english.md [hunk 6]
```diff
@@ -24,7 +24,7 @@ tests:
   - text: Delete the property <code>"tails"</code> from <code>myDog</code>.
     testString: assert(typeof myDog === "object" && myDog.tails === undefined, 'Delete the property <code>"tails"</code> from <code>myDog</code>.');
```
Comment (vkWeb) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/delete-properties-from-a-javascript-object.english.md]: This one.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/escaping-literal-quotes-in-strings.english.md [hunk 7]
```diff
@@ -28,7 +28,7 @@ tests:
   - text: You should use two double quotes (<code>&quot;</code>) and four escaped double quotes (<code>&#92;&quot;</code>).
     testString: assert(code.match(/\\"/g).length === 4 && code.match(/[^\\]"/g).length === 2, 'You should use two double quotes (<code>&quot;</code>) and four escaped double quotes (<code>&#92;&quot;</code>).');
```
Comment (vkWeb) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/escaping-literal-quotes-in-strings.english.md]: This one.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/replacing-if-else-chains-with-switch.english.md [hunk 8]
```diff
@@ -57,7 +57,7 @@ tests:
   - text: <code>chainToSwitch(42)</code> should be "The Answer"
     testString: assert(chainToSwitch(42) === "The Answer", '<code>chainToSwitch(42)</code> should be "The Answer"');
```
Comment (vkWeb) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/replacing-if-else-chains-with-switch.english.md]: Remove all the arguments in this file.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/updating-object-properties.english.md [hunk 9]
```diff
@@ -38,7 +38,7 @@ tests:
   - text: Update <code>myDog</code>&apos;s <code>"name"</code> property to equal "Happy Coder".
     testString: assert(/happy coder/gi.test(myDog.name), 'Update <code>myDog</code>&apos;s <code>"name"</code> property to equal "Happy Coder".');
```
Comment (vkWeb) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-javascript/updating-object-properties.english.md]: This one.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/debugging/catch-misspelled-variable-and-function-names.english.md [hunk 10]
```diff
@@ -21,7 +21,7 @@ Fix the two spelling errors in the code so the <code>netWorkingCapital</code> ca
 ```yml
 tests:
   - text: 'Check the spelling of the two variables used in the netWorkingCapital calculation, the console output should show that "Net working capital is: 2".'
-    testString: 'assert(netWorkingCapital === 2, ''Check the spelling of the two variables used in the netWorkingCapital calculation, the console output should show that "Net working capital is: 2".'');'
+    testString: 'assert(netWorkingCapital === 2);'
   - text: There should be no instances of mis-spelled variables in the code.
     testString: assert(!code.match(/recievables/g), 'There should be no instances of mis-spelled variables in the code.');
```
Comment (vkWeb) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/debugging/catch-misspelled-variable-and-function-names.english.md]: 4 arguments here needs to be removed.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/debugging/catch-unclosed-parentheses-brackets-braces-and-quotes.english.md [hunk 11]
```diff
@@ -23,7 +23,7 @@ tests:
   - text: Your code should fix the missing piece of the array.
     testString: assert(code.match(/myArray\s*?=\s*?\[\s*?1\s*?,\s*?2\s*?,\s*?3\s*?\];/g), 'Your code should fix the missing piece of the array.');
```
Comment (vkWeb) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/debugging/catch-unclosed-parentheses-brackets-braces-and-quotes.english.md]: This one.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-filter-method-to-extract-data-from-an-array.english.md [hunk 12]
```diff
@@ -27,7 +27,7 @@ tests:
   - text: Your code should not use a <code>for</code> loop.
     testString: assert(!code.match(/for\s*?\(.+?\)/g), 'Your code should not use a <code>for</code> loop.');
```
Comment (vkWeb) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/functional-programming/use-the-filter-method-to-extract-data-from-an-array.english.md]: 3 arguments need to be removed.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/make-a-person.english.md [hunk 13]
```diff
@@ -42,21 +42,21 @@ tests:
   - text: <code>bob.lastName</code> should return undefined.
     testString: assert.deepEqual(bob.lastName, undefined, '<code>bob.lastName</code> should return undefined.');
```
Comment (vkWeb) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/make-a-person.english.md]: 4 arguments in this file.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/missing-letters.english.md [hunk 14]
```diff
@@ -23,13 +23,13 @@ Remember to use <a href='http://forum.freecodecamp.org/t/how-to-get-help-when-yo
...
-    testString: assert.deepEqual(fearNotLetter('abcdefghjklmno'), 'i', '<code>fearNotLetter("abcdefghjklmno")</code> should return "i".');
+    testString: assert.deepEqual(fearNotLetter('abcdefghjklmno'), 'i');
   - text: <code>fearNotLetter("stvwx")</code> should return "u".
-    testString: assert.deepEqual(fearNotLetter('stvwx'), 'u', '<code>fearNotLetter("stvwx")</code> should return "u".');
+    testString: assert.deepEqual(fearNotLetter('stvwx'), 'u');
   - text: <code>fearNotLetter("bcdf")</code> should return "e".
-    testString: assert.deepEqual(fearNotLetter('bcdf'), 'e', '<code>fearNotLetter("bcdf")</code> should return "e".');
+    testString: assert.deepEqual(fearNotLetter('bcdf'), 'e');
   - text: <code>fearNotLetter("abcdefghijklmnopqrstuvwxyz")</code> should return undefined.
     testString: assert.isUndefined(fearNotLetter('abcdefghijklmnopqrstuvwxyz'), '<code>fearNotLetter("abcdefghijklmnopqrstuvwxyz")</code> should return undefined.');
```
Comment (vkWeb) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/missing-letters.english.md]: This one.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/sum-all-primes.english.md [hunk 15]
```diff
@@ -24,7 +24,7 @@ Remember to use <a href='http://forum.freecodecamp.org/t/how-to-get-help-when-yo
 ```yml
 tests:
   - text: <code>sumPrimes(10)</code> should return a number.
-    testString: assert.deepEqual(typeof sumPrimes(10), 'number', '<code>sumPrimes(10)</code> should return a number.');
+    testString: assert.deepEqual(typeof sumPrimes(10), 'number');
   - text: <code>sumPrimes(10)</code> should return 17.
     testString: assert.deepEqual(sumPrimes(10), 17, '<code>sumPrimes(10)</code> should return 17.');
```
Comment (vkWeb) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/intermediate-algorithm-scripting/sum-all-primes.english.md]: 2 arguments in this file.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects/cash-register.english.md [hunk 16]
```diff
@@ -30,15 +30,15 @@ tests:
   - text: <code>checkCashRegister(19.5, 20, [["PENNY", 1.01], ["NICKEL", 2.05], ["DIME", 3.1], ["QUARTER", 4.25], ["ONE", 90], ["FIVE", 55], ["TEN", 20], ["TWENTY", 60], ["ONE HUNDRED", 100]])</code> should return an object.
     testString: assert.deepEqual(Object.prototype.toString.call(checkCashRegister(19.5, 20, [["PENNY", 1.01], ["NICKEL", 2.05], ["DIME", 3.1], ["QUARTER", 4.25], ["ONE", 90], ["FIVE", 55], ["TEN", 20], ["TWENTY", 60], ["ONE HUNDRED", 100]])), '[object Object]', '<code>checkCashRegister(19.5, 20, [["PENNY", 1.01], ["NICKEL", 2.05], ["DIME", 3.1], ["QUARTER", 4.25], ["ONE", 90], ["FIVE", 55], ["TEN", 20], ["TWENTY", 60], ["ONE HUNDRED", 100]])</code> should return an object.');
```
Comment (vkWeb) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects/cash-register.english.md]: This one.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects/palindrome-checker.english.md [hunk 17]
```diff
@@ -48,7 +48,7 @@ tests:
   - text: <code>palindrome("1 eye for of 1 eye.")</code> should return false.
     testString: assert(palindrome("1 eye for of 1 eye.") === false, '<code>palindrome("1 eye for of 1 eye.")</code> should return false.');
```
Comment (vkWeb) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/javascript-algorithms-and-data-structures-projects/palindrome-checker.english.md]: A lot of arguments are there in this file.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/extend-constructors-to-receive-arguments.english.md [hunk 18]
```diff
@@ -52,7 +52,7 @@ tests:
   - text: <code>Dog</code> should receive an argument for <code>name</code>.
     testString: assert((new Dog('Clifford')).name === 'Clifford', '<code>Dog</code> should receive an argument for <code>name</code>.');
```
Comment (vkWeb) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/object-oriented-programming/extend-constructors-to-receive-arguments.english.md]: 3 arguments.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/match-all-non-numbers.english.md [hunk 19]
```diff
@@ -35,7 +35,7 @@ tests:
   - text: Your regex should find 12 non-digits in <code>"21 Jump Street"</code>.
     testString: assert("21 Jump Street".match(noNumRegex).length == 12, 'Your regex should find 12 non-digits in <code>"21 Jump Street"</code>.');
```
Comment (vkWeb) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/match-all-non-numbers.english.md]: A lot of arguments found.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/match-all-numbers.english.md [hunk 20]
```diff
@@ -35,7 +35,7 @@ tests:
   - text: Your regex should find 2 digits in <code>"21 Jump Street"</code>.
     testString: assert("21 Jump Street".match(numRegex).length == 2, 'Your regex should find 2 digits in <code>"21 Jump Street"</code>.');
```
Comment (vkWeb) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/match-all-numbers.english.md]: A lot of arguments here too.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/match-non-whitespace-characters.english.md [hunk 21]
```diff
@@ -34,7 +34,7 @@ tests:
   - text: Your regex should find 35 non-spaces in <code>"Men are from Mars and women are from Venus."</code>
     testString: assert("Men are from Mars and women are from Venus.".match(countNonWhiteSpace).length == 35, 'Your regex should find 35 non-spaces in <code>"Men are from Mars and women are from Venus."</code>');
   - text: 'Your regex should find 23 non-spaces in <code>"Space: the final frontier."</code>'
-    testString: 'assert("Space: the final frontier.".match(countNonWhiteSpace).length == 23, ''Your regex should find 23 non-spaces in <code>"Space: the final frontier."</code>'');'
+    testString: 'assert("Space: the final frontier.".match(countNonWhiteSpace).length == 23);'
   - text: Your regex should find 21 non-spaces in <code>"MindYourPersonalSpace"</code>
     testString: assert("MindYourPersonalSpace".match(countNonWhiteSpace).length == 21, 'Your regex should find 21 non-spaces in <code>"MindYourPersonalSpace"</code>');
```
Comment (vkWeb) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/match-non-whitespace-characters.english.md]: 4 arguments.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/match-whitespace.english.md [hunk 22]
```diff
@@ -35,7 +35,7 @@ tests:
   - text: Your regex should find eight spaces in <code>"Men are from Mars and women are from Venus."</code>
     testString: assert("Men are from Mars and women are from Venus.".match(countWhiteSpace).length == 8, 'Your regex should find eight spaces in <code>"Men are from Mars and women are from Venus."</code>');
   - text: 'Your regex should find three spaces in <code>"Space: the final frontier."</code>'
-    testString: 'assert("Space: the final frontier.".match(countWhiteSpace).length == 3, ''Your regex should find three spaces in <code>"Space: the final frontier."</code>'');'
+    testString: 'assert("Space: the final frontier.".match(countWhiteSpace).length == 3);'
   - text: Your regex should find no spaces in <code>"MindYourPersonalSpace"</code>
     testString: assert("MindYourPersonalSpace".match(countWhiteSpace) == null, 'Your regex should find no spaces in <code>"MindYourPersonalSpace"</code>');
```
Comment (vkWeb) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/regular-expressions/match-whitespace.english.md]: 4 arguments.

**Review by vkWeb — CHANGES_REQUESTED** [review]
@RandellDawson Except for a few files, all the files have message arguments that need to be removed. Please go through all the files once more.

**Comment by RandellDawson** [comment]
@vkWeb This PR actually is an extra PR which removed arguments which were missed in the series of other PRs for the JavaScript Algorithm and Data Structures section.  This PR only will contain fixes which were missed in the earlier PRs.

**Comment by vkWeb** [comment]
@RandellDawson Ohh. Okay, I'll give this one a quick look after rebase.

**Comment by thecodingaviator** [comment]
This is the only such PR left now @RandellDawson

**Review by thecodingaviator — APPROVED** [review]
LGTM

**Comment by thecodingaviator** [comment]
Thank you for your contribution to the page! 👍
We're happy to accept these changes, and look forward to future contributions. 📝

**Pull request merged**